sirepo/util.py

```diff
@@ -183,8 +183,11 @@ def err(obj, fmt='', *args, **kwargs):
 
 def flask_app():
     import flask
+    import sirepo.server
 
-    return flask.current_app or None
+    if not sirepo.server.server_initialized:
```

Comment:
Hmm let me think about this more. The problem is we've been using `flask_app()` sometimes to mean `are we in a flask request and thus have a user set` clearly that abstraction is broken. in `audit_proprietary_lib_files` I was calling it and we did have a flask_app (we had called flask.Flask()) but we did not have a user. So I added `server_initialized` which got around the problem. I think I'll add a concept to auth that figures out if we have a user set or not and call that.

Reply:
Ok, I hope I've improved the code. https://github.com/radiasoft/sirepo/pull/3339/commits/ad15177a79a5ea818de10c3d8caab9957aae01cd

`server_initialized` was never needed. I was confused about `flask.current_app`. I didn't realize it is an object that is set per request so calling `util.flask_app()` to determine if we are in a request does actually work. 

Before I made that realization I refactored the code to move the call to `set_user` outside of `audit_proprietary_lib_files` (the call to set_user is what initially made me add `server_initialized` because I thought flask_app would be True even though we weren't in a request when doing the db upgrade for the flash lib files). You were suspicious of this code initially and per usual that suspicion was correct:) 

In addition I refactored the roles code to not supply the uid. Previously we would give the uid to the db call (ex `auth_db.UserRole.add_role(uid, role)`. I think this was a code smell. We now call `auth.logged_in_user` to get the uid. In the case of pkcli.roles we call `set_user` at the outer level so the call to `logged_in_user` when accessing the db just works. In the case of add_roles during a request the user is set by the request handling code so the call to `logged_in_user` works as usual.